twitter/app/Models/Tweet.php

```diff
@@ -86,4 +94,36 @@ public function searchByQuery(string $keyword): Collection
             return $searchdTweet;
         }
     }
+
+    /**
+     * いいねしているかを調べる
+     */
+    public function isFavorite(int $userId): bool
+    {
+        return $this->favorites()->where('user_id', $userId)->exists();
+    }
+
+    /**
+     * いいねする
+     */
+    public function favorite(int $userId): void
+    {
+        if($this->isFavorite($userId)) {
+
+        } else {
+            $this->favorites()->attach($userId);
+        }
+    }
+
+    /**
+     * いいねを外す
+     */
+    public function unfavorite(int $userId): void
+    {
+        if($this->isFavorite($userId)) {
+            $this->favorites()->detach($userId);
+        } else {
+
+        }
```

Comment:
こちらも同じだけど、条件式が同じならひとつのメソッドにまとめてしまおう。
以下のようにすればunfavoriteは使わなくていいはず
```suggestion
    public function favorite(int $userId): void
    {
        if($this->isFavorite($userId)) {
            $this->favorites()->detach($userId);
        } else {
            $this->favorites()->attach($userId);
        }
```